indico/modules/legal/__init__.py

```diff
@@ -51,3 +57,33 @@ def _inject_privacy_footer(**kwargs):
     url = legal_settings.get('privacy_policy_url')
     if url or legal_settings.get('privacy_policy'):
         return render_template('legal/privacy_footer.html', url=url)
+
+
+@signals.core.app_created.connect
+def _handle_terms_changed(app, **kwargs):
+    @app.before_request
+    def _terms_if_not_accepted():
+        terms_date_str = legal_settings.get('terms_effective_date')
```

Comment:
You can add converters to a SettingsProxy. we already have `DatetimeConverter`, should be very easy to create a `DateConverter` based on that